controller/controller.go

```diff
@@ -0,0 +1,153 @@
+package controller
+
+import (
+	"context"
+	"math"
+	"time"
+
+	api "github.com/aserto-dev/go-grpc/aserto/api/v2"
+	management "github.com/aserto-dev/go-grpc/aserto/management/v2"
+	"github.com/pkg/errors"
+	"github.com/rs/zerolog"
+	"golang.org/x/sync/errgroup"
+)
+
+type CommandFunc func(context.Context, *api.Command) error
+
+type Controller struct {
+	ctx        context.Context
+	client     management.ControllerClient
+	policyName string
+	host       string
+	cfg        *Config
+	logger     *zerolog.Logger
+	handle     CommandFunc
+}
+
+type sleepResult bool
+
+const (
+	timeout                     = 1 * time.Second
+	maxBackoff                  = 600 * time.Second // max waits between retries.
+	Canceled        sleepResult = true
+	DurationReached sleepResult = false
+)
+
+func NewController(logger *zerolog.Logger, ctx context.Context, policyName, host string, cfg *Config, handler CommandFunc) (*Controller, error) {
+	if cfg.Server == nil {
+		return nil, errors.New("no server configuration provided")
+	}
+	if !cfg.Enabled {
```

Comment:
Shouldn't this check happen first? If the controller isn't even enabled we should return an error if the `Server` field is empty.